### PR TITLE
polybar: init at 3.0.4

### PR DESCRIPTION
**Pull request by afldcr**
Polybar is a nice X11 status bar in the vein of i3bar or xmobar.

https://github.com/jaagr/polybar

Ignore the source branch name-- this developer moves fairly quickly :)

##### Reviewed points

- [ ] package path fits guidelines
- [ ] package name fits guidelines
- [ ] package version fits guidelines
- [ ] package build on ARCHITECTURE
- [ ] executables tested on ARCHITECTURE
- [ ] `meta.description` is set and fits guidelines
- [ ] `meta.license` fits upstream license
- [ ] `meta.platforms` is set
- [ ] `meta.maintainers` is set
- [ ] build time only dependencies are declared in `nativeBuildInputs`
- [ ] source is fetched using the appropriate function
- [ ] phases are respected
- [ ] patches that are remotely available are fetched with `fetchpatch`

##### Possible improvements

- [ ] Is there an easier way to maintain optional dependencies?
- [ ] Is there any way to read the `config.environment` setting on NixOS?

### CODE REVIEW

**Comment by afldcr** [comment]
Fairly sure I'm read to have this merged in, provided tests start passing.

**Review by asymmetric — CHANGES_REQUESTED** [review]
Small issues, otherwise works great! :+1:

File: pkgs/applications/misc/polybar/default.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      (if githubSupport then curl          else null)
+      (if iwSupport     then wirelesstools else null)
+      (if mpdSupport    then mpd_clientlib else null)
+
+      (if i3Support || i3GapsSupport then jsoncpp else null)
+      (if i3Support then i3 else null)
+      (if i3GapsSupport then i3-gaps else null)
+    ];
+
+    nativeBuildInputs = [
```
Comment (asymmetric) [inline pkgs/applications/misc/polybar/default.nix]: I think this should simply be `buildInputs`. `nativeBuildInputs` is used when cross-compiling, which isn't the case here.

File: pkgs/applications/misc/polybar/default.nix [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+assert iwSupport     -> wirelesstools != null;
+assert mpdSupport    -> mpd_clientlib != null;
+
+assert i3Support     -> ! i3GapsSupport && jsoncpp != null && i3      != null;
+assert i3GapsSupport -> ! i3Support     && jsoncpp != null && i3-gaps != null;
+
+stdenv.mkDerivation rec {
+    name = "polybar-${version}";
+    version = "3.0.4";
+    src = fetchgit {
```
Comment (afldcr) [inline pkgs/applications/misc/polybar/default.nix]: Sadly fetchFromGitHub doesn't include submodules, which causes the build process to break.

File: pkgs/applications/misc/polybar/default.nix [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      url = "https://github.com/jaagr/polybar";
+      rev = "1f31870d43f5cd87a5529a55b1d2d3e64105e0af";
+      sha256 = "1nhj4npqhs6zy161931sbdi52gz6163lik0wri9wr122sjf90jas";
+      fetchSubmodules = true;
```
Comment (afldcr) [inline pkgs/applications/misc/polybar/default.nix]: Fixed!

**Pull request merged**